### PR TITLE
The fork2

### PR DESCRIPTION
**Pull request by treblereel**
KOGITO-4065: Make sure that SceSim Editor compiles with removed modules from errai/appformer

this issue can be tests only after this PR's get merged:

https://github.com/tiagobento/droolsjbpm-build-bootstrap/pull/4
https://github.com/tiagobento/kie-wb-common/pull/18

### CODE REVIEW

File: drools-wb-screens/drools-wb-scenario-simulation-editor/drools-wb-scenario-simulation-editor-kogito-runtime/src/main/java/org/drools/workbench/screens/scenariosimulation/webapp/client/editor/DMNDiagramEditorActivity.java [hunk 1]
```diff
@@ -17,90 +17,131 @@
...
+import org.uberfire.security.ResourceType;
+import org.uberfire.workbench.model.ActivityResourceType;
+import org.uberfire.workbench.model.bridge.Notification;
 
 @ApplicationScoped
-@WorkbenchScreen(identifier = IDENTIFIER)
-public class ScenarioSimulationEditorKogitoTestingScreen extends AbstractScenarioSimulationEditorKogitoScreen {
+@Named(DMNDiagramEditorActivity.ID)
+public class DMNDiagramEditorActivity extends AbstractActivity implements EditorActivity {
+    public static final String ID = "DMNDiagramEditor";
```
Comment (caponetto) [inline drools-wb-screens/drools-wb-scenario-simulation-editor/drools-wb-scenario-simulation-editor-kogito-runtime/src/main/java/org/drools/workbench/screens/scenariosimulation/webapp/client/editor/DMNDiagramEditorActivity.java]:
```suggestion
    public static final String ID = "ScenarioSimulationEditor";
```
Please set `ScenarioSimulationEditor`as ID.

File: drools-wb-screens/drools-wb-scenario-simulation-editor/drools-wb-scenario-simulation-editor-kogito-runtime/src/main/java/org/drools/workbench/screens/scenariosimulation/webapp/client/editor/DMNDiagramEditorActivity.java [hunk 2]
```diff
@@ -17,90 +17,131 @@
...
-import static org.drools.workbench.screens.scenariosimulation.webapp.client.editor.ScenarioSimulationEditorKogitoTestingScreen.IDENTIFIER;
+import org.uberfire.security.ResourceType;
+import org.uberfire.workbench.model.ActivityResourceType;
+import org.uberfire.workbench.model.bridge.Notification;
 
 @ApplicationScoped
-@WorkbenchScreen(identifier = IDENTIFIER)
-public class ScenarioSimulationEditorKogitoTestingScreen extends AbstractScenarioSimulationEditorKogitoScreen {
+@Named(DMNDiagramEditorActivity.ID)
+public class DMNDiagramEditorActivity extends AbstractActivity implements EditorActivity {
```
Comment (caponetto) [inline drools-wb-screens/drools-wb-scenario-simulation-editor/drools-wb-scenario-simulation-editor-kogito-runtime/src/main/java/org/drools/workbench/screens/scenariosimulation/webapp/client/editor/DMNDiagramEditorActivity.java]:
```suggestion
public class ScenarioSimulationEditorActivity extends AbstractActivity implements EditorActivity {
```
Please rename this class.

**Review by caponetto — CHANGES_REQUESTED** [review]

File: drools-wb-screens/drools-wb-scenario-simulation-editor/drools-wb-scenario-simulation-editor-kogito-runtime/src/main/webapp/test.html [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+
+<!DOCTYPE html>
+<html lang="en">
+<head>
+    <meta charset="UTF-8">
+    <title>SCESIM Editor Test</title>
+
+    <script>
+        function newDiagram() {
+            window.frames.editorFrame.contentWindow.gwtEditorBeans.get("DMNDiagramEditor").get().setContent("","");
```
Comment (caponetto) [inline drools-wb-screens/drools-wb-scenario-simulation-editor/drools-wb-scenario-simulation-editor-kogito-runtime/src/main/webapp/test.html]:
```suggestion
            window.frames.editorFrame.contentWindow.gwtEditorBeans.get("ScenarioSimulationEditor").get().setContent("","");
```
Please update this ID and others below (lines 30 and 42)

File: drools-wb-screens/drools-wb-scenario-simulation-editor/drools-wb-scenario-simulation-editor-kogito-runtime/src/main/java/org/drools/workbench/screens/scenariosimulation/webapp/client/editor/DMNDiagramEditorActivity.java (hunk 1 above)
Comment (treblereel) [reply drools-wb-screens/drools-wb-scenario-simulation-editor/drools-wb-scenario-simulation-editor-kogito-runtime/src/main/java/org/drools/workbench/screens/scenariosimulation/webapp/client/editor/DMNDiagramEditorActivity.java]: Yeap, sure

File: drools-wb-screens/drools-wb-scenario-simulation-editor/drools-wb-scenario-simulation-editor-kogito-runtime/src/main/java/org/drools/workbench/screens/scenariosimulation/webapp/client/editor/DMNDiagramEditorActivity.java (hunk 2 above)
Comment (treblereel) [reply drools-wb-screens/drools-wb-scenario-simulation-editor/drools-wb-scenario-simulation-editor-kogito-runtime/src/main/java/org/drools/workbench/screens/scenariosimulation/webapp/client/editor/DMNDiagramEditorActivity.java]: ok

File: drools-wb-screens/drools-wb-scenario-simulation-editor/drools-wb-scenario-simulation-editor-client/pom.xml [hunk 4]
```diff
@@ -85,6 +85,15 @@
...
     </dependency>
 
+    <dependency>
+      <groupId>org.uberfire</groupId>
+      <artifactId>appformer-client-api</artifactId>
+    </dependency>
+    <dependency>
+      <groupId>org.drools</groupId>
+      <artifactId>drools-wb-scenario-simulation-editor-api</artifactId>
+    </dependency>
```
Comment (caponetto) [inline drools-wb-screens/drools-wb-scenario-simulation-editor/drools-wb-scenario-simulation-editor-client/pom.xml]: Please remove this duplicated dependency. We already have this one on line 74.

File: drools-wb-screens/drools-wb-scenario-simulation-editor/drools-wb-scenario-simulation-editor-kogito-client/pom.xml [hunk 5]
```diff
@@ -121,6 +121,12 @@
       <groupId>org.kie.workbench</groupId>
       <artifactId>kie-wb-common-kogito-client</artifactId>
     </dependency>
+    <dependency>
+      <groupId>org.drools</groupId>
+      <artifactId>drools-wb-scenario-simulation-editor-client</artifactId>
+      <version>7.53.0-SNAPSHOT</version>
+      <scope>compile</scope>
+    </dependency>
```
Comment (caponetto) [inline drools-wb-screens/drools-wb-scenario-simulation-editor/drools-wb-scenario-simulation-editor-kogito-client/pom.xml]: Please remove this duplicated dependency. We already have this one on line 42.

**Comment by caponetto** [comment]
I'm not sure the reason but I had to rename the package of the `SortTable` class. Otherwise, I can't compile.
```diff
-package org.drools.workbench.services.verifier.plugin.client.api;
+package org.guvnor.common.services.shared.exceptions;
```
Maybe something with the merge after integrating https://github.com/tiagobento/drools-wb/pull/1/commits/21178fbf4eeaf5986943df34c44eeb2e63b6dbe0?

File: drools-wb-screens/drools-wb-scenario-simulation-editor/drools-wb-scenario-simulation-editor-kogito-runtime/src/main/webapp/test.html [hunk 6]
```diff
@@ -0,0 +1,82 @@
...
+            window.frames.editorFrame.contentWindow.importDMN();
+            document.getElementById("download_btn").disabled = false;
+        }
+
+        function download() {
+                window.frames.editorFrame.contentWindow.gwtEditorBeans.get("ScenarioSimulationEditor").get().getContent().then(
+                    function(p) {
+                        var d = document.createElement('a');
+                        d.setAttribute('href', 'data:text/xml;charset=utf-8,' + encodeURIComponent(p));
+                        d.setAttribute('download', "diagram.sce");
```
Comment (caponetto) [inline drools-wb-screens/drools-wb-scenario-simulation-editor/drools-wb-scenario-simulation-editor-kogito-runtime/src/main/webapp/test.html]:
```suggestion
                        d.setAttribute('download', "diagram.scesim");
```
Please fix the extension name.

File: drools-wb-screens/drools-wb-scenario-simulation-editor/drools-wb-scenario-simulation-editor-kogito-runtime/pom.xml [hunk 7]
```diff
@@ -633,6 +404,131 @@
...
+          <optimizationLevel>9</optimizationLevel>
+          <style>OBFUSCATED</style>
+          <logLevel>INFO</logLevel>
+          <noServer>false</noServer>
+          <!--
+                        <server>org.jboss.errai.cdi.server.gwt.EmbeddedWildFlyLauncher</server>
+          -->
+          <runTarget>test.html</runTarget>
+          <strict>true</strict>
+          <style>PRETTY</style>
```
Comment (caponetto) [inline drools-wb-screens/drools-wb-scenario-simulation-editor/drools-wb-scenario-simulation-editor-kogito-runtime/pom.xml]: Please remove this duplicated tag. Line 416 already sets it.

**Review by caponetto — APPROVED** [review]

**Pull request merged**